### PR TITLE
Adding Unify AI as an LLM Provider

### PR DESCRIPTION
**Pull request by paras55**
Added Support for [Unify AI](https://docs.unify.ai/basics/welcome) along with cookbook recipe.
- Phidata Users can now use all the Models (Open Sourced and Closed Sourced) from all the providers with just 1 single API key 

```
from phi.assistant import Assistant
from phi.tools.duckduckgo import DuckDuckGo
from phi.llm.unify import UnifyChat

assistant = Assistant(
    llm=UnifyChat(
        endpoint="gpt-4o@openai",  # or claude-3-sonnet@anthropic
        api_key="",  
    ),
    tools=[DuckDuckGo()],
    show_tool_calls=True,
)

assistant.print_response("Five learnings from Bhagwad Geeta?", markdown=True, stream=False)   
```

### CODE REVIEW

File: phi/llm/unify/unify.py [hunk 1]
```diff
@@ -0,0 +1,204 @@
+from typing import Optional, List, Iterator, Dict, Any
+from phi.llm.base import LLM
+from phi.llm.message import Message
+from phi.utils.log import logger
+from phi.utils.timer import Timer
+
+try:
+    from unify.clients import Unify as UnifyClient, AsyncUnify as AsyncUnifyClient
```
Comment (ysolanky) [inline phi/llm/unify/unify.py:8]: `from unify import Unify as UnifyClient, AsyncUnify as AsyncUnifyClient`

**Comment by ysolanky** [comment]
Hey @paras55. Thanks for your contribution! Though even after the fix above, the `UnifyChat` class is not working. 

I am running into the error: `Exception: You must provider either the user_message or messages!` which suggests that the invoke function needs to be customized

**Comment by paras55** [comment]
Hi @ysolanky Thanks for the PR review :) 

I pushed the fix. Please let me know if there is anything that should be changed/updated.

**Comment by jacobweiss2305** [comment]
@paras55 getting this error:
```
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 401 Client Error: Unauthorized for url: https://api.unify.ai/v0/endpoints
```
for this chunk of code:
```
from phi.assistant import Assistant
from phi.tools.duckduckgo import DuckDuckGo
from phi.llm.unify import UnifyChat

import os

assistant = Assistant(
    llm=UnifyChat(
        endpoint="gpt-4o@openai", 
        api_key=os.getenv("OPENAI_API_KEY"),  
    ),
    tools=[DuckDuckGo()],
    show_tool_calls=True,
)

assistant.print_response("Five learnings from Bhagwad Geeta?", markdown=True, stream=False)   
```

I added my OpenAI key to :
Provider Keys
Query supported endpoints directly with your personal provider keys.

Am I missing something?

**Comment by jacobweiss2305** [comment]
Closing, add to feature requests

**Pull request closed without merging**